opm/autodiff/SimulatorBase_impl.hpp

```diff
@@ -61,15 +62,13 @@ namespace Opm
             allcells_[cell] = cell;
         }
 #if HAVE_MPI
-        if ( terminal_output_ ) {
-            if ( solver_.parallelInformation().type() == typeid(ParallelISTLInformation) )
-            {
-                const ParallelISTLInformation& info =
-                    boost::any_cast<const ParallelISTLInformation&>(solver_.parallelInformation());
-                // Only rank 0 does print to std::cout
-                terminal_output_ = ( info.communicator().rank() == 0 );
-                is_parallel_run_ = ( info.communicator().size() > 1 );
-            }
+        if ( solver_.parallelInformation().type() == typeid(ParallelISTLInformation) )
+        {
+            const ParallelISTLInformation& info =
+                boost::any_cast<const ParallelISTLInformation&>(solver_.parallelInformation());
+            // Only rank 0 does print to std::cout
+            terminal_output_ &= ( info.communicator().rank() == 0 );
```

Comment:
What exactly do you mean with you can't. I tested it and it worked. Do you mean it might fail with different compilers? Anyway I'll change it (and merge it) because it's easier to read anyway.